File: inventory-service/src/main/java/com/linktic/inventory/service/InventoryService.java
package com.linktic.inventory.service;

import com.linktic.inventory.dto.InventoryDTO;
import com.linktic.inventory.dto.PageResponse;
import com.linktic.inventory.dto.PurchaseRequest;
import com.linktic.inventory.dto.PurchaseResultDTO;
import com.linktic.inventory.dto.UpdateInventoryRequest;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface InventoryService {
    PageResponse<InventoryDTO> findAll(Pageable pageable);
    InventoryDTO findByProductoId(UUID productoId);
    InventoryDTO update(UUID productoId, UpdateInventoryRequest request);
    PurchaseResultDTO purchase(PurchaseRequest request);
}

File: frontend/src/services/inventoryService.js
import { inventoryApi } from './api.js'

export const inventoryService = {
  async getAll(page = 0, size = 10) {
    const { data } = await inventoryApi.get('/api/v1/inventory', { params: { page, size } })
    return data.data
  },

  async getByProductId(productoId) {
    const { data } = await inventoryApi.get(`/api/v1/inventory/${productoId}`)
    return data.data
  },

  async update(productoId, payload) {
    const { data } = await inventoryApi.put(`/api/v1/inventory/${productoId}`, payload)
    return data.data
  },

  async purchase(payload) {
    const { data } = await inventoryApi.post('/api/v1/purchases', payload)
    return data.data
  }
}
